### PR TITLE
Clippy fixes

### PR DESCRIPTION
**Pull request by ckaran**
This commit used clippy to find and fix a bunch things.  See the commit messages for what got fixed.

### CODE REVIEW

File: src/core/events.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl PartialOrd for EventKey {
 
 impl Ord for EventKey {
     fn cmp(&self, other: &Self) -> Ordering {
-        if self.time == other.time {
+        if (self.time - other.time).abs() < std::f64::EPSILON {
```
Comment (SergiusIW) [inline src/core/events.rs]: Exact equality is expected in this if statement. It is for a comparison tiebreaker.

**Review by SergiusIW — CHANGES_REQUESTED** [review]
Concerning the "Fixed float_cmp error" commit. Exact equality is expected in all of these cases. These equality checks should not have a tolerance associated with them. I will not merge the "Fixed float_cmp error" commit, as it may introduce unexpected behaviour.

**Comment by ckaran** [comment]
OK, do you want me to revert those changes and make a new pull request?

**Comment by SergiusIW** [comment]
Yes please. The other clippy fixes were fine.

**Comment by ckaran** [comment]
OK, I've made the changes.  I'm not sure how GitHub handles updates to a pull request; do I close this one and create a new pull request, or do I just leave this as-is and wait for you to merge the new changes in?

**Comment by SergiusIW** [comment]
You can make the changes in this pull request. You should be able to just revert the float_cmp commit and push to the clippy_fixes branch on ckaran/collider-rs.

**Comment by ckaran** [comment]
And I've learned something new today! :smile: I didn't know that this was how you fixed a pull request, nor did I know about the `git revert` command.  Thank you for both!  Please let me know if the fixes pass muster.

**Pull request merged**